src/services/platform.services.js

```diff
@@ -0,0 +1,47 @@
+import { PACKAGE_MANAGER } from '../config/app';
+const childProcess = window.require('child_process');
+const os = window.require('os');
+const path = window.require('path');
+const { remote } = window.require('electron');
+
+// Returns true if the OS is Windows
+export const isWin = (): boolean => /^win/.test(os.platform());
+
+// Returns path to the users Documents direactory
+export const getWindowsHomeDir = (): string => {
+  // For Windows Support
+  // Documents folder is much better place for project folders (Most programs use it as a default save location)
+  // Since there is a chance of being moved or users language might be differet we are reading the value from Registery
+  // There might be a better solution but this seems ok so far
+
+  const winDocumentsRegRecord = childProcess.execSync(
+    'REG QUERY "HKCU\\Software\\Microsoft\\Windows\\CurrentVersion\\Explorer\\User Shell Folders" /v Personal',
+    {
+      encoding: 'utf8',
+    }
+  );
+
+  const winDocPathArray = winDocumentsRegRecord.split(' ');
+  const winDocPath = winDocPathArray[winDocPathArray.length - 1]
+    .replace('%USERPROFILE%\\', '')
+    .replace(/\s/g, '');
+  return path.join(os.homedir(), winDocPath);
+};
+
+// Returns formatted command for Windows
+export const formatCommandForPlatform = (command: string): string =>
+  isWin() ? `${command}.cmd` : command;
+
+// Returns PATH for Windows
+export const getPathForPlatform = (): string =>
+  isWin()
+    ? `${remote.app.getPath(
+        'appData'
+      )}\\npm;C:\\Program Files\\nodejs;C:\\Program Files (x86)\\Yarn\\bin`
```

Comment:
`x86` refers to a 32-bit OS, right?

I believe I recall that 32-bit programs can still run in 64-bit OS', and so this may be the install path for both 32-bit and 64-bit operating systems?

I wonder if on 32-bit systems, it would just be `Program Files\Yarn\bin`...

Reply:
Good catch. The x86 is also available on a 64bit Windows for 32 bit apps. Yes, on 32bit it's probably with-out x86 in path. I think this also depends on how you installed Yarn. At my machine I don't have it at that location, I installed with `npm i -g yarn` and not with an installer.

Anyway, I think we can remove yarn from env. path because we're using the yarn that's installed with Guppy (see PACKAGE_MANAGER_CMD).